frontend/src/pages/MilestonesEditor.tsx

```diff
@@ -363,4 +371,12 @@ export const MilestonesEditor = () => {
       </div>
     </DragDropContext>
   );
+
+  const renderLoading = () => (
+    <div className={classes(css.loadingSpinnerContainer)}>
+      <LoadingSpinner />
+    </div>
+  );
+
+  return areInitialVersions ? renderLoading() : renderPage();
```

Comment:
Well you could get rid of the functions `renderPage `and `renderLoading` and simply handle everything in the return. Less code but might be a bit harder to read so might not be worth it though. Maybe if the `areInitialVersions` was called something like `isLoading` here then it could be more readable
```suggestion

  return areInitialVersions ? (
    <div className={classes(css.loadingSpinnerContainer)}>
      <LoadingSpinner />
    </div>
  ) : (
    <DragDropContext onDragEnd={onDragEnd}>
      <InfoTooltip
        title={
          <div>
            <Trans i18nKey="Planner milestones tooltip">
              Milestones create your project’s
              <Link to={`/roadmap/${roadmapId}/planner/graph`}>roadmap.</Link>
            </Trans>
          </div>
        }
      >
        <div className={classes(css.infoBackground)}>
          <InfoIcon className={classes(css.tooltipIcon, css.infoIcon)} />
        </div>
      </InfoTooltip>
      <div className={classes(css.layoutRow, css.overflowYAuto)}>
        <ExpandableColumn
          className={classes(css.unorderedCol)}
          expanded={expandUnordered}
          onToggle={() => setExpandUnordered((prev) => !prev)}
          title={
            <div>
              {`${t('Unordered tasks')} (${
                versionLists[ROADMAP_LIST_ID]?.length ?? 0
              })`}
            </div>
          }
        >
          <SortableTaskList
            listId={ROADMAP_LIST_ID}
            tasks={versionLists[ROADMAP_LIST_ID] || []}
            disableDragging={disableDrag}
            showRatings
            showSearch
          />
        </ExpandableColumn>
        {versionLists && renderMilestones()}
      </div>
    </DragDropContext>
  );
```

Reply:
Even better (for readability), return early if areInitialVersions is true
```suggestion
  if (areInitialVersions)
    <div className={classes(css.loadingSpinnerContainer)}>
      <LoadingSpinner />
    </div>;
  return (
    <DragDropContext onDragEnd={onDragEnd}>
      <InfoTooltip
        title={
          <div>
            <Trans i18nKey="Planner milestones tooltip">
              Milestones create your project’s
              <Link to={`/roadmap/${roadmapId}/planner/graph`}>roadmap.</Link>
            </Trans>
          </div>
        }
      >
        <div className={classes(css.infoBackground)}>
          <InfoIcon className={classes(css.tooltipIcon, css.infoIcon)} />
        </div>
      </InfoTooltip>
      <div className={classes(css.layoutRow, css.overflowYAuto)}>
        <ExpandableColumn
          className={classes(css.unorderedCol)}
          expanded={expandUnordered}
          onToggle={() => setExpandUnordered((prev) => !prev)}
          title={
            <div>
              {`${t('Unordered tasks')} (${
                versionLists[ROADMAP_LIST_ID]?.length ?? 0
              })`}
            </div>
          }
        >
          <SortableTaskList
            listId={ROADMAP_LIST_ID}
            tasks={versionLists[ROADMAP_LIST_ID] || []}
            disableDragging={disableDrag}
            showRatings
            showSearch
          />
        </ExpandableColumn>
        {versionLists && renderMilestones()}
      </div>
    </DragDropContext>
  );
```

Reply:
I originally tried to put everything in return with ternary, but prettier-linter insisted that has to be made look super confusing, so I ditched that and made own functions for them.

This is the version that the linter likes.

![image](https://user-images.githubusercontent.com/101560117/159344824-826d2fdc-1cd3-403d-a20a-0d256a65b032.png)

At least to me that looks less clear, but it might look better for others.

And yeah, 'isLoading' could make more sense at this point.

Reply:
Oh, you already posted this version 😄

Reply:
Something like this would be pretty clear to me, but the linter gets super mad. Its not as dense, but I think it shows more clearly what each of these ternary operations results would do.

![image](https://user-images.githubusercontent.com/101560117/159348677-8aafe1ea-4381-4386-942c-1b11bd2d6138.png)

But then again, maybe I'm a freak and this is actually a horrible design 😄

Reply:
Iidas version looks best to me. I don't know why her message took so long to be visible for me.

But thats very readable, probably gonna go with that one.

Reply:
Unless, could it be confusing that the "official" return might not happen at all? Or is that fine?

Reply:
> Unless, could it be confusing that the "official" return might not happen at all? Or is that fine?

The same pattern is often used in this project, so it's okay 👍

Reply:
Returning early does have some drawbacks but overall its the pattern I prefer

https://medium.com/swlh/return-early-pattern-3d18a41bba8